### ISSUE TITLE
Windows: Game crashes with "Can't create an OpenGL window"

### ISSUE THREAD

**Issue by Costpap** [a-b-street/abstreet#103]
Hello. After updating to [v0.2.0e](https://github.com/dabreegster/abstreet/releases/download/v0.2.0a/abstreet_windows_v0_2_0e.zip), upon launching the game via `play_abstreet.bat`, the game will not launch with error "Can't create an OpenGL window". Below is a copy of my `output.txt` file, with `RUST_BACKTRACE` set to `full`.

```
thread 'main' panicked at 'Can't create an OpenGL window. Please file an issue at https://github.com/dabreegster/abstreet/issues/ and include output.txt. First attempt Couldn't find any pixel format that matches the criteria., second attempt Couldn't find any pixel format that matches the criteria.', ezgui\src\backend_glium.rs:32:26
stack backtrace:
   0:     0x7ff7658602df - <unknown>
   1:     0x7ff76587f49b - <unknown>
   2:     0x7ff765857a6c - <unknown>
   3:     0x7ff7658635ec - <unknown>
   4:     0x7ff76586323f - <unknown>
   5:     0x7ff765863dc7 - <unknown>
   6:     0x7ff76586394f - <unknown>
   7:     0x7ff7658638bc - <unknown>
   8:     0x7ff7656796cd - <unknown>
   9:     0x7ff76507e58b - <unknown>
  10:     0x7ff764fd3646 - <unknown>
  11:     0x7ff76513bd26 - <unknown>
  12:     0x7ff7658637e7 - <unknown>
  13:     0x7ff76586ee12 - <unknown>
  14:     0x7ff765864118 - <unknown>
  15:     0x7ff764fd4507 - <unknown>
  16:     0x7ff7658f2e14 - <unknown>
  17:     0x7ff89ee86fd4 - BaseThreadInitThunk
  18:     0x7ff8a02bcec1 - RtlUserThreadStart

```

**Comment by dabreegster** [comment]
Oh no, declared victory too soon. Could you please run http://realtech-vr.com/home/glview and send the .xml file it produces? We can see what pixel formats your videocard supports, and make a third fallback option that'll work.

**Comment by Costpap** [comment]
Sure, I'll send the .xml file in a few minutes.

*Note: I have a really old intergrated graphics card, so I wouldn't be suprized if it's a hardware issue. Also, I'll be upgrading in a few weeks time, so I'm not really sure if it's worth fixing this issue.*

**Comment by Costpap** [comment]
I assume that this is what you need, so here you go:

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<!-- http://www.realtech-vr.com/glview -->
<GLView version="6.0.5"><System>Windows 10 x64 Edition  (Build 9200)</System>
<Vendor>Microsoft Corporation</Vendor>
<Version>1.1.0</Version>
<DirectXVersion>DirectX 9, 10.0</DirectXVersion>
<DXShaderModel>vs_4_0, ps_4_0</DXShaderModel>
<Renderer>GDI Generic</Renderer>
<Extensions>GL_WIN_swap_hint GL_EXT_bgra GL_EXT_paletted_texture</Extensions>
<VRAM>1771</VRAM><GPUClock>0</GPUClock><Capabilities>
<FrameBuffer>
<ViewportMaxWidth>16384</ViewportMaxWidth>
<ViewportMaxHeight>16384</ViewportMaxHeight>
<SubPixelBits>3</SubPixelBits>
</FrameBuffer>
<Stack>
<AttribStackDepth>16</AttribStackDepth>
<NameStackDepth>128</NameStackDepth>
<MaxModelViewStackDepth>32</MaxModelViewStackDepth>
<MaxProjectionStackDepth>10</MaxProjectionStackDepth>
<MaxClientAttribStackDepth>16</MaxClientAttribStackDepth>
<MaxTextureStackDepth>10</MaxTextureStackDepth>
</Stack>
<Various>
<LineSizeMin>0.5</LineSizeMin>
<LineSizeMax>10</LineSizeMax>
<LineSizeGran>0.125</LineSizeGran>
<PointSizeMin>0.5</PointSizeMin>
<PointSizeMax>10</PointSizeMax>
<PointSizeGran>0.125</PointSizeGran>
<AliasedLineWidthMin>0</AliasedLineWidthMin>
<AliasedLineWidthMax>0</AliasedLineWidthMax>
<AliasedPointSizeMin>0</AliasedPointSizeMin>
<AliasedPointSizeMax>0</AliasedPointSizeMax>
<NumClipPlanes>6</NumClipPlanes>
<NumLights>8</NumLights>
<PixelMapTable>65536</PixelMapTable>
<TextureSize>1024</TextureSize>
<EvalOrder>30</EvalOrder>
</Various>
<Extensions_Specs>
<MaxTextureAnisotropy>0</MaxTextureAnisotropy>
<MaxTextureUnits>0</MaxTextureUnits>
<MaxTextureCubeSize>0</MaxTextureCubeSize>
<NumCompressedTextureFormats>0</NumCompressedTextureFormats>
<MaxVertexBlend>0</MaxVertexBlend>
<MaxTexture3DSize>0</MaxTexture3DSize>
<MaxOcclusionQueryBits>0</MaxOcclusionQueryBits>
<ATI></ATI><nVidia></nVidia></Extensions_Specs></Capabilities><ICD>
<name></name>
<company></company>
<file></file>
<version>0</version>
<description></description>
</ICD>
<PixelFormats><p id="1" color_bits="32" depth_bits="32" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="True" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="2" color_bits="32" depth_bits="16" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="True" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="3" color_bits="32" depth_bits="32" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="True" support_opengl="True" draw_to_window="True" draw_to_bitmap="False" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="4" color_bits="32" depth_bits="16" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="True" support_opengl="True" draw_to_window="True" draw_to_bitmap="False" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="5" color_bits="32" depth_bits="32" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="True" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="6" color_bits="32" depth_bits="16" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="True" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="7" color_bits="32" depth_bits="32" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="True" support_opengl="True" draw_to_window="True" draw_to_bitmap="False" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="8" color_bits="32" depth_bits="16" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="True" support_opengl="True" draw_to_window="True" draw_to_bitmap="False" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="9" color_bits="32" depth_bits="32" stencil_bits="8" accum_bits="0" pixel_type="1" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="True" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="10" color_bits="32" depth_bits="16" stencil_bits="8" accum_bits="0" pixel_type="1" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="True" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="11" color_bits="32" depth_bits="32" stencil_bits="8" accum_bits="0" pixel_type="1" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="True" support_opengl="True" draw_to_window="True" draw_to_bitmap="False" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="12" color_bits="32" depth_bits="16" stencil_bits="8" accum_bits="0" pixel_type="1" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="True" support_opengl="True" draw_to_window="True" draw_to_bitmap="False" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="13" color_bits="24" depth_bits="32" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="14" color_bits="24" depth_bits="16" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="15" color_bits="24" depth_bits="32" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="16" color_bits="24" depth_bits="16" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="17" color_bits="24" depth_bits="32" stencil_bits="8" accum_bits="0" pixel_type="1" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="18" color_bits="24" depth_bits="16" stencil_bits="8" accum_bits="0" pixel_type="1" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="19" color_bits="16" depth_bits="32" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="20" color_bits="16" depth_bits="16" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="21" color_bits="16" depth_bits="32" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="22" color_bits="16" depth_bits="16" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="23" color_bits="16" depth_bits="32" stencil_bits="8" accum_bits="0" pixel_type="1" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="24" color_bits="16" depth_bits="16" stencil_bits="8" accum_bits="0" pixel_type="1" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="25" color_bits="8" depth_bits="32" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="26" color_bits="8" depth_bits="16" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="27" color_bits="8" depth_bits="32" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="28" color_bits="8" depth_bits="16" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="29" color_bits="8" depth_bits="32" stencil_bits="8" accum_bits="0" pixel_type="1" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="30" color_bits="8" depth_bits="16" stencil_bits="8" accum_bits="0" pixel_type="1" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="31" color_bits="4" depth_bits="32" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="32" color_bits="4" depth_bits="16" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="33" color_bits="4" depth_bits="32" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="34" color_bits="4" depth_bits="16" stencil_bits="8" accum_bits="0" pixel_type="0" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="35" color_bits="4" depth_bits="32" stencil_bits="8" accum_bits="0" pixel_type="1" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /><p id="36" color_bits="4" depth_bits="16" stencil_bits="8" accum_bits="0" pixel_type="1" transparent="False" acceleration="True" support_gdi="False" aux_buffers="0" double_buffer="False" support_opengl="True" draw_to_window="False" draw_to_bitmap="True" swap_layers_buffers="0" stereo="False" auxbuffers="0" samples="0" number_overlays="0" number_underlays="0" type_rgba_float="False" type_rgba_unsigned_float="False" framebuffer_srgb_capable="False"  /></PixelFormats>
<Resolutions>
<Mode index="0" width="320" height="200" depth="32" rate="70"/>
<Mode index="1" width="320" height="240" depth="32" rate="70"/>
<Mode index="2" width="400" height="300" depth="32" rate="70"/>
<Mode index="3" width="512" height="384" depth="32" rate="70"/>
<Mode index="4" width="640" height="400" depth="32" rate="70"/>
<Mode index="5" width="640" height="480" depth="32" rate="59"/>
<Mode index="6" width="640" height="480" depth="32" rate="60"/>
<Mode index="7" width="640" height="480" depth="32" rate="70"/>
<Mode index="8" width="640" height="480" depth="32" rate="73"/>
<Mode index="9" width="640" height="480" depth="32" rate="75"/>
<Mode index="10" width="640" height="480" depth="32" rate="85"/>
<Mode index="11" width="640" height="480" depth="32" rate="100"/>
<Mode index="12" width="640" height="480" depth="32" rate="120"/>
<Mode index="13" width="800" height="600" depth="32" rate="60"/>
<Mode index="14" width="800" height="600" depth="32" rate="72"/>
<Mode index="15" width="800" height="600" depth="32" rate="75"/>
<Mode index="16" width="800" height="600" depth="32" rate="85"/>
<Mode index="17" width="800" height="600" depth="32" rate="100"/>
<Mode index="18" width="800" height="600" depth="32" rate="120"/>
<Mode index="19" width="1024" height="768" depth="32" rate="60"/>
<Mode index="20" width="1024" height="768" depth="32" rate="70"/>
<Mode index="21" width="1024" height="768" depth="32" rate="75"/>
<Mode index="22" width="1024" height="768" depth="32" rate="85"/>
<Mode index="23" width="1024" height="768" depth="32" rate="100"/>
<Mode index="24" width="1024" height="768" depth="32" rate="120"/>
<Mode index="25" width="1152" height="864" depth="32" rate="60"/>
<Mode index="26" width="1152" height="864" depth="32" rate="75"/>
<Mode index="27" width="1152" height="864" depth="32" rate="85"/>
<Mode index="28" width="1152" height="864" depth="32" rate="100"/>
<Mode index="29" width="1280" height="600" depth="32" rate="60"/>
<Mode index="30" width="1280" height="720" depth="32" rate="60"/>
<Mode index="31" width="1280" height="720" depth="32" rate="75"/>
<Mode index="32" width="1280" height="720" depth="32" rate="85"/>
<Mode index="33" width="1280" height="720" depth="32" rate="100"/>
<Mode index="34" width="1280" height="768" depth="32" rate="60"/>
<Mode index="35" width="1280" height="768" depth="32" rate="75"/>
<Mode index="36" width="1280" height="768" depth="32" rate="85"/>
<Mode index="37" width="1280" height="768" depth="32" rate="120"/>
<Mode index="38" width="1280" height="800" depth="32" rate="60"/>
<Mode index="39" width="1280" height="800" depth="32" rate="75"/>
<Mode index="40" width="1280" height="800" depth="32" rate="85"/>
<Mode index="41" width="1280" height="800" depth="32" rate="120"/>
<Mode index="42" width="1280" height="960" depth="32" rate="60"/>
<Mode index="43" width="1280" height="960" depth="32" rate="75"/>
<Mode index="44" width="1280" height="960" depth="32" rate="85"/>
<Mode index="45" width="1280" height="960" depth="32" rate="120"/>
<Mode index="46" width="1280" height="1024" depth="32" rate="60"/>
<Mode index="47" width="1280" height="1024" depth="32" rate="75"/>
<Mode index="48" width="1280" height="1024" depth="32" rate="85"/>
<Mode index="49" width="1280" height="1024" depth="32" rate="100"/>
<Mode index="50" width="1280" height="1024" depth="32" rate="120"/>
<Mode index="51" width="1360" height="768" depth="32" rate="60"/>
<Mode index="52" width="1360" height="768" depth="32" rate="120"/>
<Mode index="53" width="1366" height="768" depth="32" rate="60"/>
<Mode index="54" width="1400" height="1050" depth="32" rate="60"/>
<Mode index="55" width="1400" height="1050" depth="32" rate="75"/>
<Mode index="56" width="1400" height="1050" depth="32" rate="85"/>
<Mode index="57" width="1400" height="1050" depth="32" rate="120"/>
<Mode index="58" width="1440" height="900" depth="32" rate="60"/>
<Mode index="59" width="1440" height="900" depth="32" rate="75"/>
<Mode index="60" width="1440" height="900" depth="32" rate="85"/>
<Mode index="61" width="1440" height="900" depth="32" rate="120"/>
<Mode index="62" width="1600" height="900" depth="32" rate="60"/>
<Mode index="63" width="1600" height="900" depth="32" rate="75"/>
<Mode index="64" width="1600" height="900" depth="32" rate="85"/>
<Mode index="65" width="1600" height="900" depth="32" rate="100"/>
<Mode index="66" width="1600" height="900" depth="32" rate="120"/>
<Mode index="67" width="1600" height="1200" depth="32" rate="60"/>
<Mode index="68" width="1600" height="1200" depth="32" rate="70"/>
<Mode index="69" width="1600" height="1200" depth="32" rate="75"/>
<Mode index="70" width="1600" height="1200" depth="32" rate="85"/>
<Mode index="71" width="1600" height="1200" depth="32" rate="100"/>
<Mode index="72" width="1600" height="1200" depth="32" rate="120"/>
<Mode index="73" width="1680" height="1050" depth="32" rate="60"/>
<Mode index="74" width="1680" height="1050" depth="32" rate="75"/>
<Mode index="75" width="1680" height="1050" depth="32" rate="85"/>
<Mode index="76" width="1680" height="1050" depth="32" rate="120"/>
<Mode index="77" width="1792" height="1344" depth="32" rate="60"/>
<Mode index="78" width="1792" height="1344" depth="32" rate="75"/>
<Mode index="79" width="1856" height="1392" depth="32" rate="60"/>
<Mode index="80" width="1856" height="1392" depth="32" rate="75"/>
<Mode index="81" width="1920" height="1080" depth="32" rate="60"/>
<Mode index="82" width="1920" height="1080" depth="32" rate="75"/>
<Mode index="83" width="1920" height="1080" depth="32" rate="85"/>
<Mode index="84" width="1920" height="1080" depth="32" rate="100"/>
<Mode index="85" width="1920" height="1200" depth="32" rate="60"/>
<Mode index="86" width="1920" height="1200" depth="32" rate="75"/>
<Mode index="87" width="1920" height="1200" depth="32" rate="85"/>
<Mode index="88" width="1920" height="1440" depth="32" rate="60"/>
<Mode index="89" width="1920" height="1440" depth="32" rate="75"/>
<Mode index="90" width="1920" height="1440" depth="32" rate="85"/>
<Mode index="91" width="2048" height="1152" depth="32" rate="60"/>
<Mode index="92" width="2048" height="1536" depth="32" rate="60"/>
<Mode index="93" width="2048" height="1536" depth="32" rate="75"/>
<Mode index="94" width="2048" height="1536" depth="32" rate="93"/>
<Mode index="95" width="2560" height="1600" depth="32" rate="60"/>
</Resolutions>
</GLView>

```

**Comment by dabreegster** [comment]
https://github.com/dabreegster/abstreet/releases/download/v0.2.0a/abstreet_windows_v0_2_0f.zip
6 attempts at a working Windows build later... :P

**Comment by Costpap** [comment]
Hmmm. Still getting the same issue: [output.txt](https://github.com/dabreegster/abstreet/files/4837005/output.txt)

**Comment by dabreegster** [comment]
I'm kind of stumped. The only other idea I have is to change how `glutin` picks the pixel format to use, adding a mode to just accept anything offered. I don't think I'll be able to prioritize this anytime soon, though -- really sorry. :\

**Comment by Costpap** [comment]
Hmm. As I said before, this might be a hardware issue. Considering how old my system is, I wouldn't be suprized. You could maybe consider adding minimum and recommended requirements so that people know if the game can run on their system or not :P

Feel free to close this issue if you want to.

**Comment by dabreegster** [comment]
I'll keep the bug open to
1) try out the glutin "take anything we can get" idea
2) figure out and document the system requirements

**Comment by Costpap** [comment]
Ah, okay. Sounds good to me.

**Comment by dabreegster** [comment]
https://github.com/a-b-street/abstreet/blob/5f3c914e28f7e51eeacd4edc5119cf0ad1366d14/widgetry/src/backend_glow_native.rs#L22
We have some more fallback cases here. Please reopen if you're still having problems with the latest release.

**Comment by Costpap** [comment]
Unfortunately, I no longer regularly use the PC with which I was able to reproduce this bug. While I do still have access to it, I do not use it for anything. Since then, I have upgraded to a newer PC, to which I later on added a discrete GPU to, and the last time I tried to play the game on the newer iGPU, it worked fine. While in theory I could test it, I find it unlikely that the fallbacks won't work and that anyone else who will attempt to run the game will have such an old GPU that doesn't support what are essentially modern technologies.
